### PR TITLE
dbp 929 linux server for email

### PR DESCRIPTION
**Pull request by jdabat**
# Description
added module for creating a lan and firewall rules

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: modules/ionos-lan-and-firewall/main.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Create a LAN ( private or public )
+ resource "ionoscloud_lan" "lan" {
+   datacenter_id = var.datacenter_id
+   public        = var.is_public
+   name          = format( "%s-%s",var.name,"intern-connection-lan")
```
Comment (dimapin) [inline modules/ionos-lan-and-firewall/main.tf:5]: The suffix intern-connection-lan contradicts the setting public-true. Why don't you use the name which is given with the variable

File: modules/ionos-lan-and-firewall/main.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ }
+
+# Create a NIC for the basic VM and connect it to the LAN
+resource "ionoscloud_nic" "nic" {
+  datacenter_id   = var.datacenter_id
+  server_id       = var.server_id
+  dhcp            = true
```
Comment (dimapin) [inline modules/ionos-lan-and-firewall/main.tf:12]: This should be optional

Comment (jdabat) [reply modules/ionos-lan-and-firewall/main.tf:12]: This module is not fully implemented, still missing parts. Currently the needed parts are implemented. extending the module is still a TODO

File: modules/ionos-lan-and-firewall/main.tf [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Create a LAN ( private or public )
+ resource "ionoscloud_lan" "lan" {
+   datacenter_id = var.datacenter_id
+   public        = var.is_public
+   name          = format( "%s-%s-%x",var.name,var.is_public? "public":"private","-connection-lan")
```
Comment (dimapin) [inline modules/ionos-lan-and-firewall/main.tf:5]: The format should be %s-%s-%s

**Review by dimapin — CHANGES_REQUESTED** [review]

**Review by dimapin — APPROVED** [review]

**Pull request merged**